tasks/modules.py

```diff
@@ -403,3 +405,42 @@ def validate(_: Context):
         message += "Please add them to the DEFAULT_MODULES list or exclude them from the validation."
 
         raise Exit(message)
+
+
+@task
+def validate_used_by_otel(ctx: Context):
+    """
+    Verify whether indirect local dependencies of modules labeled "used_by_otel" are also marked with the "used_by_otel" tag.
+    """
+    otel_mods = [path for path, module in DEFAULT_MODULES.items() if module.used_by_otel]
+    missing_used_by_otel_label: dict[str, list[str]] = defaultdict(list)
+
+    # for every module labeled as "used_by_otel"
+    for otel_mod in otel_mods:
+        gomod_path = f"{otel_mod}/go.mod"
+        # get the go.mod data
+        result = ctx.run(f"go mod edit -json {gomod_path}", hide='both')
+        if result.failed:
+            raise Exception(f"Error running go mod edit -json on {gomod_path}: {result.stderr}")
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ ‏Good practice for invoke is to use the `Exit` class
```
from invoke.exceptions import Exit
from tasks.libs.common.color import Color, color_message
```
```suggestion
        if result.failed:
            raise Exception()
            raise Exit(color_message(f"Error running go mod edit -json on {gomod_path}: {result.stderr}", Color.RED), code=1)
```